### PR TITLE
ci: codedev token

### PR DESCRIPTION
**Pull request by zirain**
fix: https://github.com/envoyproxy/gateway/pull/2558

doc: https://docs.codecov.com/docs/adding-the-codecov-token#github-actions

### CODE REVIEW

**Comment by zirain** [comment]
@phlax @Xunzhuo need token for codedev, can you help to set the secret?

**Comment by Xunzhuo** [comment]
Will update it tomorrow

**Comment by zirain** [comment]
> Will update it tomorrow

not sure why it's passed now.

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -52,8 +52,9 @@ jobs:
     - name: Run Coverage Tests
       run: make go.test.coverage
     - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@4fe8c5f003fae66aa5ebb77cfd3e7bfbbda0b6b0  # v3.1.5
+      uses: codecov/codecov-action@e0b68c6749509c5f83f984dd99a76a1c1a231044  # v4.0.1
       with:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (arkodg) [inline .github/workflows/build_and_test.yaml:57]: how did this token get injected into secrets ?

Comment (zirain) [reply .github/workflows/build_and_test.yaml:57]: i have no idea why it passed now, maybe  there is a org secret.

Comment (phlax) [reply .github/workflows/build_and_test.yaml:57]: i would assume that no password is being passed - i think that is standard for opensource/public/prs - and dont see any secret defined anywhere

Comment (arkodg) [reply .github/workflows/build_and_test.yaml:57]: we do have the codecov app installed for the repo, which maybe related

Comment (phlax) [reply .github/workflows/build_and_test.yaml:57]: yeah - its installed as an app - so it gets notifications and probs ci results from that

**Comment by Xunzhuo** [comment]
Should we close this ?

**Comment by arkodg** [comment]
> Should we close this ?

yeah, I crosschecked and we are not adding a TOKEN to our repo or org secrets, so not sure what's happening here

**Pull request closed without merging**